src/app/helpers/terra-tooltip-helper.directive.ts

```diff
@@ -0,0 +1,115 @@
+import {
+    Directive,
+    ElementRef,
+    EventEmitter,
+    HostListener,
+    Input,
+    OnInit,
+    Output
+} from '@angular/core';
+import { StringHelper } from './string.helper';
+
+@Directive(
+    {
+        selector: '[terraTooltipHelper]'
+    }
+)
+export class TerraTooltipHelperDirective implements OnInit
+{
+    @Input()
+    public nameOfInput:string;
+
+    @Input()
+    public tooltipTextOfInput:string;
+
+    @Output()
+    public tooltipOfInputDisabledChange:EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    @Output()
+    public tooltipDisabledChange:EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    @Output()
+    public tooltipTextChange:EventEmitter<string> = new EventEmitter<string>();
+
+    private _tooltipOfInputDisabled:boolean;
+
+    private _tooltipDisabled:boolean;
+
+    private _tooltipText:string;
+
+    @Input()
+    public get tooltipOfInputDisabled():boolean
+    {
+        return this._tooltipOfInputDisabled;
+    }
+
+    public set tooltipOfInputDisabled(value:boolean)
+    {
+        this._tooltipOfInputDisabled = value;
+        this.tooltipOfInputDisabledChange.emit(value);
+    }
+
+    @Input()
+    public get tooltipDisabled():boolean
+    {
+        return this._tooltipDisabled;
+    }
+
+    public set tooltipDisabled(value:boolean)
+    {
+        this._tooltipDisabled = value;
+        this.tooltipDisabledChange.emit(value);
+    }
+
+    @Input()
```

Comment:
wie kann der Input ein Getter sein?

Reply:
haaa määäächick XDDDD

https://blog.angulartraining.com/tutorial-create-your-own-two-way-data-binding-in-angular-46487650ea82

Resumen:

ist für's 2-way-data-binding ^^